pandas/core/arrays/text.py

```diff
@@ -0,0 +1,279 @@
+import operator
+from typing import TYPE_CHECKING, Type
+
+import numpy as np
+
+from pandas._libs import lib
+
+from pandas.core.dtypes.base import ExtensionDtype
+from pandas.core.dtypes.common import pandas_dtype
+from pandas.core.dtypes.dtypes import register_extension_dtype
+from pandas.core.dtypes.generic import ABCDataFrame, ABCIndexClass, ABCSeries
+from pandas.core.dtypes.inference import is_array_like
+
+from pandas import compat
+from pandas.core import ops
+from pandas.core.arrays import PandasArray
+from pandas.core.construction import extract_array
+from pandas.core.missing import isna
+
+if TYPE_CHECKING:
+    from pandas._typing import Scalar
+
+
+@register_extension_dtype
+class TextDtype(ExtensionDtype):
+    """
+    Extension dtype for text data.
+
+    .. versionadded:: 1.0.0
+
+    .. warning::
+
+       TextDtype is considered experimental. The implementation and
+       parts of the API may change without warning.
+
+       In particular, TextDtype.na_value may change to no longer be
+       ``numpy.nan``.
+
+    Attributes
+    ----------
+    None
+
+    Methods
+    -------
+    None
```

Comment:
are `__repr__` and construct_from_string not included here?  More generally, if there is nothing to list here, is the section needed?